functions/sendgrid/index.js

```diff
@@ -1,101 +1,105 @@
+/**
+ * Copyright 2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 'use strict';
 
-// [START setup]
-var async = require('async');
-var sendgrid = require('sendgrid');
-var config = require('./config.json');
-var gcloud = require('google-cloud');
-var uuid = require('node-uuid');
+// [START functions_sendgrid_setup]
+const sendgrid = require('sendgrid');
+const config = require('./config.json');
+const uuid = require('node-uuid');
 
 // Get a reference to the Cloud Storage component
-var storage = gcloud.storage();
+const storage = require('@google-cloud/storage')();
 // Get a reference to the BigQuery component
-var bigquery = gcloud.bigquery();
-// [END setup]
+const bigquery = require('@google-cloud/bigquery')();
+// [END functions_sendgrid_setup]
 
-// [START getClient]
+// [START functions_sendgrid_get_client]
 /**
  * Returns a configured SendGrid client.
  *
  * @param {string} key Your SendGrid API key.
- * @returns {Object} SendGrid client.
+ * @returns {object} SendGrid client.
  */
 function getClient (key) {
   if (!key) {
-    var error = new Error('SendGrid API key not provided. Make sure you have a ' +
-      '"sg_key" property in your request querystring');
+    const error = new Error('SendGrid API key not provided. Make sure you have a "sg_key" property in your request querystring');
     error.code = 401;
     throw error;
   }
 
   // Using SendGrid's Node.js Library https://github.com/sendgrid/sendgrid-nodejs
-  return sendgrid.SendGrid(key);
+  return sendgrid(key);
 }
-// [END getClient]
+// [END functions_sendgrid_get_client]
 
-// [START getPayload]
+// [START functions_get_payload]
 /**
  * Constructs the SendGrid email request from the HTTP request body.
  *
- * @param {Object} requestBody Cloud Function request body.
+ * @param {object} requestBody Cloud Function request body.
  * @param {string} data.to Email address of the recipient.
  * @param {string} data.from Email address of the sender.
  * @param {string} data.subject Email subject line.
  * @param {string} data.body Body of the email subject line.
- * @returns {Object} Payload object.
+ * @returns {object} Payload object.
  */
 function getPayload (requestBody) {
   if (!requestBody.to) {
-    var error = new Error('To email address not provided. Make sure you have a ' +
-      '"to" property in your request');
+    const error = new Error('To email address not provided. Make sure you have a "to" property in your request');
```

Comment:
1) too much abstraction
2) I have to first save the error to a variable so I can add the `code` property to it.